Sources/NIOCore/ByteBuffer-quicBinaryEncodingStrategy.swift

```diff
@@ -0,0 +1,138 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2024 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+extension ByteBuffer {
+    public struct QUICBinaryEncodingStrategy: NIOBinaryIntegerEncodingStrategy {
+        public var requiredBytesHint: Int
+
+        /// - Parameter requiredBytesHint: An estimate of the bytes required to write integers using this strategy
+        @inlinable
+        public init(requiredBytesHint: Int) {
```

Comment:
Do we think it's useful to make this configurable?

Reply:
In some cases, a user might have a rough idea of how many bytes they're expecting to write. In such cases, allowing them to use a smaller number would result in saved bytes

Reply:
An alternative would be, make this not configurable, but then add a parameter to ByteBuffer/writeLengthPrefixed(strategy:writeData:)

Reply:
leaning towards the alternative
Having the requiredBytesHint parameter here is quite confusing